vanguard.py

```diff
@@ -2,68 +2,121 @@
 
 def do_SILSLA_13(record):
-	change_CLU(record)
-	delete_752(record)
+    change_CLU(record)
+    delete_752(record)
 
-#SILSLA-14
+# SILSLA-14
 
 def delete_956(record):
-	"""Delete all 956 fields"""
-	for fld in record.get_fields('956'):
-		record.remove_field(fld)
+    """Delete all 956 fields"""
+    for fld in record.get_fields("956"):
+        record.remove_field(fld)
 
 def copy_856(record):
-	"""Copy contents of 856 into new 956 field"""
-	for fld in record.get_fields('856'):
-		fld_956 = copy.copy(fld)
-		fld_956.tag = '956'
-		record.add_ordered_field(fld_956)	
+    """Copy contents of 856 into new 956 field"""
+    for fld in record.get_fields("856"):
+        fld_956 = copy.copy(fld)
+        fld_956.tag = "956"
+        record.add_ordered_field(fld_956)
 
 def do_SILSLA_14(record):
-	delete_956(record)
-	copy_856(record)
+    delete_956(record)
+    copy_856(record)
 
+# SILSLA-15
+
+def delete_various_9xx(record):
+    """Delete various 9xx fields"""
+    for fld in record.get_fields(
+        "996", "966", "951", "916", "920", "992", "962", "949"
+    ):
+        record.remove_field(fld)
+
+def get_dbcode(filename):
+    """Helper function to get dbcode from filename"""
+    for dbcode in ["filmntvdb", "ucladb", "ethnodb"]:
+        if filename.startswith(dbcode):
+            return dbcode
+        else:
+            raise ValueError(
+                f"Usage: {filename} filename must include originating dbcode"
+            )
+
+def copy_001(record, filename):
+    """Copy 001 field, append dbcode, and add as $a to 996"""
+    dbcode = get_dbcode(filename)
+    for fld in record.get_fields("001"):
+        voyager_code = "{}-{}".format(copy.copy(fld.value()), dbcode)
+        fld_996 = Field(tag="996", indicators=[' ',' '], subfields=['a', voyager_code])
+        record.add_ordered_field(fld_996)
+
+def move_9xx(record):
+    """Move various 9xx fields"""
+    field_mapping = {"901": "966", "910": "951", "935": "992", "948": "962"}
+    for old_field in field_mapping.keys():
+        new_field = field_mapping[old_field]
+        for fld in record.get_fields(old_field):
+            fld.tag = new_field
+            record.add_ordered_field(fld)
+
+def move_939_fatadb(record):
+    """For FATADB records, remove 969 field and move 939 to 969"""
+    for old_fld in record.get_fields("969"):
+        record.remove_field(old_fld)
+    for fld in record.get_fields("939"):
+        fld.tag = "969"
+        record.add_ordered_field(fld)
+
+def do_SILSLA_15(record, filename):
+    delete_various_9xx(record)
+    copy_001(record, filename)
+    move_9xx(record)
+    
 if len(sys.argv) != 3:
-    raise ValueError(f'Usage: {sys.argv[0]} in_file out_file')
+    raise ValueError(f"Usage: {sys.argv[0]} in_file out_file")
 
-reader = MARCReader(open(sys.argv[1], 'rb'))
-writer = MARCWriter(open(sys.argv[2], 'wb'))
+reader = MARCReader(open(sys.argv[1], "rb"))
+writer = MARCWriter(open(sys.argv[2], "wb"))
+dbcode = get_dbcode(sys.argv[1])
```

Comment:
Since you're now getting `dbcode` here, in the main routine, you can pass it into the various functions instead of a filename, and remove the `get_dbcode` call from those.

That also should clean up the repeated use of `sys.argv[1]`, which otherwise should be assigned to a meaningful variable for reuse.  That won't be needed after switching to `dbcode`, I think.

Reply:
Thanks, I've made those changes.